### PR TITLE
[UXE-5558] feat: migrate digital certificate to v4

### PR DESCRIPTION
**Pull request by pauloSF0**
### What is the new behavior introduced by this PR?
[UXE-5558] feat: migrate digital certificate to v4
### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/user-attachments/assets/1ff2c488-1f3b-43a3-bd42-3ddd4b5acea0

### Does it have a link on Figma?

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

[UXE-5558]: https://aziontech.atlassian.net/browse/UXE-5558?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aloisio-m-bastian — APPROVED** [review]

**Review by peterpaulo-azion — APPROVED** [review]

File: src/views/DigitalCertificates/ListView.vue [hunk 1]
```diff
@@ -39,9 +40,9 @@
   import Illustration from '@/assets/svg/illustration-layers.vue'
   import ContentBlock from '@/templates/content-block'
   import EmptyResultsBlock from '@/templates/empty-results-block'
-  import ListTableBlock from '@/templates/list-table-block'
   import { columnBuilder } from '@/templates/list-table-block/columns/column-builder'
   import PageHeadingBlock from '@/templates/page-heading-block'
+  import FetchListTableBlock from '@/templates/list-table-block/with-fetch-ordering-and-pagination.vue'
```
Comment (lucasmendes21) [inline src/views/DigitalCertificates/ListView.vue:45]: Acho que poderia deixar o componente como ListTableBlock pois a ideia é que depois esse componente de fetch morra.

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const subjectNames = item.subject_name.map((subject) => subject)?.join(',')
+    const typeMap = {
+      edge_certificate: EDGE_CERTIFICATE,
+      trusted_ca_certificate: TRUSTED_CA_CERTIFICATE
+    }
+    return {
+      id: item.id,
+      name: item.name,
+      issuer: item.issuer || '-',
+      subjectName: subjectNames === '' ? '-' : subjectNames,
```
Comment (lucasmendes21) [inline src/services/digital-certificates-services/v4/list-digital-certificates-service.js:67]: Sugestão  `subjectName: subjectNames ?? '-' ,`

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      type: typeMap[item.certificate_type] || '-',
+      validity: item.validity === null ? '-' : parseValidityDate(item.validity),
```
Comment (lucasmendes21) [inline src/services/digital-certificates-services/v4/list-digital-certificates-service.js:69]: `validity: item.validity ? parseValidityDate(item.validity) : '-'`

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    const typeMap = {
+      edge_certificate: EDGE_CERTIFICATE,
+      trusted_ca_certificate: TRUSTED_CA_CERTIFICATE
+    }
+    return {
+      id: item.id,
+      name: item.name,
+      issuer: item.issuer || '-',
+      subjectName: subjectNames === '' ? '-' : subjectNames,
+      type: typeMap[item.certificate_type] || '-',
```
Comment (lucasmendes21) [inline src/services/digital-certificates-services/v4/list-digital-certificates-service.js:68]: `typeMap[item.certificate_type] ?? '-'`

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const parsedDomains = httpResponse.body.results?.map((item) => {
+    const subjectNames = item.subject_name.map((subject) => subject)?.join(',')
+    const typeMap = {
+      edge_certificate: EDGE_CERTIFICATE,
+      trusted_ca_certificate: TRUSTED_CA_CERTIFICATE
+    }
+    return {
+      id: item.id,
+      name: item.name,
+      issuer: item.issuer || '-',
```
Comment (lucasmendes21) [inline src/services/digital-certificates-services/v4/list-digital-certificates-service.js:66]: `issuer: item.issuer ?? '-',`

File: src/views/DigitalCertificates/ListView.vue (hunk 1 above)
Comment (pauloSF0) [reply src/views/DigitalCertificates/ListView.vue:45]: Caso ele volte, ai ele será adicionado novamente

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js (hunk 3 above)
Comment (pauloSF0) [reply src/services/digital-certificates-services/v4/list-digital-certificates-service.js:69]: Esse aqui irei deixar, pois, é uma validação especifica para quando for null

File: src/services/digital-certificates-services/v4/list-digital-certificates-service.js (hunk 4 above)
Comment (pauloSF0) [reply src/services/digital-certificates-services/v4/list-digital-certificates-service.js:68]: O que está implementado é aconselhável igual a sugestão

**Pull request merged**